src/tests/class_tests/openms/source/SpectrumExtractor_test.cpp

```diff
@@ -583,6 +584,77 @@ START_SECTION(annotateSpectrum())
 }
 END_SECTION
 
+START_SECTION(scoreSpectrum())
+{
+  MzMLFile mzml;
+  PeakMap experiment;
+  TargetedExperiment targeted_exp;
+  TransitionTSVReader tsv_reader;
+
+  ptr->setRTWindow(30);
+  ptr->setMZTolerance(0.1);
+  ptr->setGaussWidth(0.25);
+  ptr->setUseGauss(true);
+
+  mzml.load(OPENMS_GET_TEST_DATA_PATH("SpectrumExtractor_13C1.mzML"), experiment);
+  std::vector<MSSpectrum> spectra = experiment.getSpectra();
+  tsv_reader.convertTSVToTargetedExperiment(OPENMS_GET_TEST_DATA_PATH("SpectrumExtractor_13CFlux_TraML.csv"), FileTypes::CSV, targeted_exp);
+  std::vector<MSSpectrum> annotated;
+
+  ptr->annotateSpectrum(spectra, targeted_exp, annotated);
+  std::vector<MSSpectrum> picked;
+  picked.resize(annotated.size());
+
+  for (UInt j=0; j<annotated.size(); ++j)
+  {
+    ptr->pickSpectrum(annotated[j], picked[j]);
+  }
+
+  std::vector<MSSpectrum> scored;
+  ptr->scoreSpectrum(annotated, picked, scored);
+
+  // std::sort(scored.begin(), scored.end(), [](MSSpectrum a, MSSpectrum b)
+  // {
+  //   return a.getFloatDataArrays()[1][0] > b.getFloatDataArrays()[1][0];
+  // });
+  std::sort(scored.begin(), scored.end(), [](MSSpectrum a, MSSpectrum b)
+  {
+    return a.getName().compare(b.getName()) < 0;
+  });
+
+  for (auto a : scored)
+  {
+    cout << a.getName()
+    << "\tscore: " << a.getFloatDataArrays()[1][0]
+    << "\tlog10_tic: " << a.getFloatDataArrays()[2][0]
+    << "\t1/fwhm: " << a.getFloatDataArrays()[3][0]
+    << "\tSNR: " << a.getFloatDataArrays()[4][0] << endl;
+  }
+
+  map<string,MSSpectrum> transition_best_spec;
+  for (auto a : scored)
+  {
+    String transition_name = a.getName();
+    map<string,MSSpectrum>::const_iterator it = transition_best_spec.find(transition_name);
+    if (it == transition_best_spec.end())
+    {
+      transition_best_spec.insert({transition_name, a});
+    }
+    else if (it->second.getFloatDataArrays()[1][0] < a.getFloatDataArrays()[1][0])
+    {
+      transition_best_spec.erase(transition_name);
+      transition_best_spec.insert({transition_name, a});
+    }
+  }
```

Comment:
Elegant solution to picking out the best scoring spectrum for each transition.  This will look nice when you incorporate this into the class.